uniffi_bindgen/src/bindings/swift/gen_swift/mod.rs

```diff
@@ -272,6 +272,27 @@ impl Config {
     }
 }
 
+// Given a trait, work out what the protocol name we generate for it.
+// This differs based on whether the trait supports foreign impls (ie,
+// whether is has a "callback interface".
+fn trait_protocol_name(ci: &ComponentInterface, name: &str) -> Result<String> {
+    let (obj_name, has_callback_interface) = match ci.get_object_definition(name) {
+        Some(obj) => (obj.name(), obj.has_callback_interface()),
+        None => (
+            ci.get_callback_interface_definition(name)
+                .ok_or_else(|| anyhow::anyhow!("no interface {}", name))?
+                .name(),
+            true,
+        ),
+    };
+    let class_name = SwiftCodeOracle.class_name(obj_name);
+    if has_callback_interface {
+        Ok(class_name)
+    } else {
+        Ok(format!("{class_name}Protocol"))
```

Comment:
Is there a good reason why this has "Protocol" appended?  IIUC, this is a Rust trait without callback interface support, so Swift can't implement it.  In that case, it seems like protocol should get `{class_name}` which feels like a nicer name, and the class that implements the protocol should get something like`{class_name}Impl`.  I think that's how the other languages handle it.

Reply:
> Is this also your understanding - ie, that this patch doesn't change the class hierarchy for existing types
Yes agreed.

> ..and that doing so is something we'd need to be very careful about?

Maybe I have a slight difference of opinion here.  I think it could be okay to make the change, since most end-users aren't going to see a difference.    If they store a `Foo` and only use it to call methods from the protocol, it shouldn't matter if `Foo` is a concrete type that implements the protocol or the protocol itself.

I agree we should be careful, but maybe I'd remove the "very" part.

Reply:
Also, I think I got confused and believed that we had already made the change for other languages.  If Kotlin works this way, then I think Swift should too.  If we decide to change them, let's change both at the same time.